### PR TITLE
feat: stacked APY rewards -- LF-17256

### PR DESCRIPTION
**Pull request by laurentsenta**
Contributes to https://lifi.atlassian.net/browse/LF-17256

Sister PR: https://github.com/jumperexchange/jumper-backend/pull/674

- [x] Update API
- [x] Adds stacked chart (specialized on rewards for now) - [storybook](https://jumper-exchange-storybook-git-feat-diffe-9162e3-jumper-exchange.vercel.app/?path=/story/core-charts-stackedareachart--default)
- [x] Refactor how data and theme are moved around for consistency
- [x] Tweak format number, add test, and fix its type
- [x] Provide a hook to get currently active scheme

## Note

I'm adding `useResolvedColorScheme` to save on mistakes when we try to access the theme, and `useIsLightTheme` which we use a lot.

That'll prevent cases where we mix dark and light scheme because mode is actually system:

## Test:

Visit opportunities, check the APY charts, the rewards should show up as a stacked chart (it might be 0%)
Examples with rewards:

http://localhost:3000/earn/moonwell-flagship-usdc-on-base
http://localhost:3000/earn/gauntlet-usdc-prime-on-base

## Summary by CodeRabbit

* **New Features**
  * Added APY analytics visualization with configurable time ranges (daily, weekly, monthly, yearly)
  * Introduced enhanced charting capabilities to display APY and TVL analytics with improved visual clarity

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/components/core/charts/StackedAreaChart/StackedAreaChart.tsx [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+            dataKey="base"
+            stroke={theme.baseLineColor}
+            fillOpacity={0}
+            fill="transparent"
+            isAnimationActive
+            activeDot={false}
+            style={{
+              transform: AREA_CONFIG.TRANSFORM,
+            }}
+          />
```
Comment (mmioana) [inline src/components/core/charts/StackedAreaChart/StackedAreaChart.tsx:319]: Nice solution 👍

File: src/components/core/charts/StackedAreaChart/index.ts
Comment (mmioana) [inline src/components/core/charts/StackedAreaChart/index.ts:1]: This should be removed

**Review by mmioana — APPROVED** [review]
Amazing work once again @laurentsenta 🙌 
Please run `pnpm run test:snapshots:generate` before merging this 😬

File: src/components/core/charts/StackedAreaChart/StackedAreaTooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      })}
+    >
+      <Typography
+        variant="bodySmallStrong"
+        style={{ textTransform: 'capitalize' }}
+      >
+        {formatDateLocalized(label ?? '', 'PP @ HH:mm')}
+      </Typography>
+      <Typography variant="bodySmall">
+        Total: {formatValue(total)} <strong>APY</strong>
```
[the remark this answers is not in the document]
Comment (laurentsenta) [reply src/components/core/charts/StackedAreaChart/StackedAreaTooltip.tsx:73]: This is temporary (three strike refactor'ish rule)

File: src/components/core/charts/StackedAreaChart/StackedAreaChart.tsx [hunk 3]
```diff
@@ -129,7 +129,7 @@ export const StackedAreaChart = ({
 
   const valueFormatter = useCallback(
     (value: number) => {
-      if (!value || isNaN(value) || value === 0) {
+      if (isNaN(value)) {
```
Comment (mmioana) [inline src/components/core/charts/StackedAreaChart/StackedAreaChart.tsx:118]: I think for `0` or empty values we still need this check so we don't format the value, but indeed we don't need both `!value` and `value === 0`

**Review by mmioana — COMMENTED** [review]
Also while testing this, I've noticed there are some duplicates among the Y axis, is this expected?

https://jumper-exchange-git-feat-differentiate-a-6afba1-jumper-exchange.vercel.app/earn/aave-v3-wsteth-on-mainnet

Image
Image

**Comment by laurentsenta** [comment]
> Also while testing this, I've noticed there are some duplicates among the Y axis, is this expected?

Those are not duplicates, it's different percentages without enough precision on the rendering. I've reworked the code to use the current formulas right now (render a 0 - 100% range when APY is constant)

**Review by callMeTheQA — APPROVED** [review]
LGTM. 
well done! 
https://github.com/user-attachments/assets/6f3a536e-460d-4eb5-87d1-a180dfdc6623
